application/common/tests/test_dynamodb.py

```diff
@@ -0,0 +1,81 @@
+from pytest import fixture
+from os import environ
+from json import dumps, loads
+from decimal import Decimal
+from common.dynamodb import (
+    add_change_request_to_dynamodb,
+    get_latest_sequence_id_for_a_given_odscode_from_dynamodb,
+    dict_hash,
+    TTL,
+)
+from boto3.dynamodb.types import TypeDeserializer
+from time import time
+
+
+@fixture
+def dynamodb_table_create(dynamodb_client):
+    """Create a DynamoDB CHANGE_EVENTS_TABLE table fixture."""
+    table = dynamodb_client.create_table(
+        TableName=environ["CHANGE_EVENTS_TABLE_NAME"],
+        BillingMode="PAY_PER_REQUEST",
+        KeySchema=[
+            {"AttributeName": "Id", "KeyType": "HASH"},
+            {"AttributeName": "ODSCode", "KeyType": "RANGE"},
+        ],
+        AttributeDefinitions=[
+            {"AttributeName": "Id", "AttributeType": "S"},
+            {"AttributeName": "ODSCode", "AttributeType": "S"},
+            {"AttributeName": "SequenceNumber", "AttributeType": "N"},
+        ],
+        GlobalSecondaryIndexes=[
+            {
+                "IndexName": "gsi_ods_sequence",
+                "KeySchema": [
+                    {"AttributeName": "ODSCode", "KeyType": "HASH"},
+                    {"AttributeName": "SequenceNumber", "KeyType": "RANGE"},
+                ],
+                "Projection": {"ProjectionType": "ALL"},
+            }
+        ],
+    )
+    return table
+
+
+def test_add_change_request_to_dynamodb(dynamodb_table_create, change_event, dynamodb_client):
+    # Arrange
+    event_received_time = int(time())
+    # Act
+    change_id = dict_hash(change_event)
+    response_add = add_change_request_to_dynamodb(change_event.copy(), 1, str(event_received_time))
+
+    item = dynamodb_client.get_item(
+        TableName=environ["CHANGE_EVENTS_TABLE_NAME"],
+        Key={"Id": {"S": change_id}, "ODSCode": {"S": change_event["ODSCode"]}},
+    )["Item"]
+    deserializer = TypeDeserializer()
+    deserialized = {k: deserializer.deserialize(v) for k, v in item.items()}
+    expected = loads(dumps(change_event), parse_float=Decimal)
+
+    assert response_add["ResponseMetadata"]["HTTPStatusCode"] == 200
+    assert deserialized["EventReceived"] == str(event_received_time)
+    assert deserialized["TTL"] == str(event_received_time + TTL)
+    assert deserialized["Id"] == change_id
+    assert deserialized["SequenceNumber"] == 1
+    assert deserialized["Event"] == expected
+
+
+def test_get_latest_sequence_id_for_a_given_odscode_from_dynamodb(dynamodb_table_create, change_event):
+    event_received_time = int(time())
+    add_change_request_to_dynamodb(change_event.copy(), 1, str(event_received_time))
```

Comment:
currently the partition key is the hash of the message so using the same change event for each will result in 1 row being updated rather than multiple rows being added for the same change event, you should have a test where there is different events for the same ods code, only need to be slightly different, then check this still works, you might find that by default your query then returns the lowest rather than the highest but this can then be easily resolved using ScanIndexForward parameter in your query .. 

`Query results are always sorted by the sort key value. If the data type of the sort key is Number, the results are returned in numeric order; otherwise, the results are returned in order of UTF-8 bytes. By default, the sort order is ascending. To reverse the order, set the ScanIndexForward parameter to false.`

https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html